### ISSUE TITLE
Templates (CSS / JS / etc.)

### ISSUE THREAD

**Issue by karbowiak** [EVE-KILL/zKillboard#249]
Rework the entire template system a bit.

So each template can come with it's own css / js / img that is then optimized into a single file and cached for subsequent loads.

If debug is true, it's just "cat'ed" together on each load
If debug isn't true, it'll go through the entire caching system

It will obv. add a slight increase in load time, because it has to check if the files changed every so often, plus the cached data itself will timeout every x hours.
But if done correctly, we can eliminate pretty much all of the public/ directory. If not the entire directory outright.

So to sum up
1. Move everything from public to the template directory itself (maybe rename template to theme)
2. Create caching system that both caches and optimizes, but also disables parts of itself when debugging is on
3. Change so a specific 'theme' can be default, instead of the current bootstrap default
